### ISSUE TITLE
it should be possible to redefine a computed property

### ISSUE THREAD

**Issue by mweststrate** [mobxjs/mobx#1121]
Example;

```javascript
const a = observable({
  width: 10, 
  get surface() { return this.width }
})

observe(a, "surface", v => console.log(v))
reaction(() => a.surface, v => console.log(v))

extendObservable(a, {
  get surface() { return this.width * 2 }
})
```

Should no longer throw but work as expected.

Downside: there might be issues with lingering low-level subscriptions, the above `observe` based logger will keep listing to the 'old' definition (which is the reason it is now forbidden). Normal property access like the `reaction` should work correctly. Potential confusion about the difference is the reason it has not been supported so far

**Comment by mweststrate** [comment]
Fixed through #1151, will be part of next release

**Comment by mweststrate** [comment]
Fixed as part of 3.3.0
